client/state/imports/reducer.js

```diff
@@ -0,0 +1,290 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import { flow, curry, omit, omitBy, includes, get } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { createReducer } from 'state/utils';
+import {
+	IMPORTS_STORE_RESET,
+	IMPORTS_FETCH,
+	IMPORTS_FETCH_FAILED,
+	IMPORTS_FETCH_COMPLETED,
+	IMPORTS_IMPORT_START,
+	IMPORTS_START_IMPORTING,
+	IMPORTS_IMPORT_CANCEL,
+	IMPORTS_IMPORT_RESET,
+	IMPORTS_IMPORT_RECEIVE,
+	IMPORTS_UPLOAD_START,
+	IMPORTS_UPLOAD_FAILED,
+	IMPORTS_UPLOAD_SET_PROGRESS,
+	IMPORTS_UPLOAD_COMPLETED,
+	IMPORTS_AUTHORS_START_MAPPING,
+	IMPORTS_AUTHORS_SET_MAPPING,
+	IMPORTS_IMPORT_LOCK,
+	IMPORTS_IMPORT_UNLOCK,
+} from 'state/action-types';
+import { appStates } from 'state/imports/constants';
+
+/**
+ * The inital state of the imports store
+ *
+ * @type {Object}
+ */
+const initialState = {
+	count: 0,
+	importers: {},
+	importerLocks: {},
+	api: {
+		isHydrated: false,
+		isFetching: false,
+		retryCount: 0,
+	},
+};
+
+/**
+ * Increment the count property of the store state.
+ *
+ * @param {Object} state The state of the redux store
+ * @returns {Object} state The updated state with count incremented by 1
+ */
+const incrementCount = state => ( { ...state, count: state.count + 1 } );
```

Comment:
To make reducers simpler, we often slice them into individual pieces and combine them with `combineReducers`. For example, the `count` reducer could look like this:
```js
const count = ( state = 0, action ) => {
  switch ( action.type ) {
    case IMPORTS_STORE_RESET:
      return 0;
    case IMPORTS_IMPORT_START:
      return state + 1;
  }
  return state;
}

export default combineReducers( count, ... );
```
Very clean, isn't? And it can also avoid many nested updates.